.github/workflows/make_release.yaml

```diff
@@ -7,99 +7,108 @@ on:
         description: 'Version of the new release, just as a number with no prepended "v"'
         required: true
 
+env:
+  PYTHON_VERSION: 3.8
+  NEW_VERSION: ${{ inputs.version }}
+
 jobs:
-  increment_version:
+  increment-version:
+    name: Bump project version and commit it
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: Install Python tools
+        uses: BrandonLWhite/pipx-install-action@v0.1.1
+      - uses: actions/setup-python@v5
         with:
-          python-version: "3.9"
-      - name: Run version script
+          python-version: ${{ env.PYTHON_VERSION }}
+          cache: poetry
+
+      - name: Install dependencies
+        run: poetry install --only=release
+
+      - name: Bump project version
         id: script
-        run: |
-          python extra/release.py bump "${{ inputs.version }}"
+        run: poe bump "${{ env.NEW_VERSION }}"
+
       - uses: EndBug/add-and-commit@v9
         name: Commit the changes
         with:
-          message: "Increment version numbers to ${{ inputs.version }}"
+          message: "Increment version to ${{ env.NEW_VERSION }}"
 
   build:
+    name: Get changelog and build the distribution package
     runs-on: ubuntu-latest
-    needs: increment_version
+    needs: increment-version
     steps:
       - uses: actions/checkout@v4
+      - name: Install Python tools
+        uses: BrandonLWhite/pipx-install-action@v0.1.1
+      - uses: actions/setup-python@v5
         with:
-          fetch-depth: 0
-          ref: master
-      - name: Set up Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.9"
-      - run: pip install build wheel sphinx
+          python-version: ${{ env.PYTHON_VERSION }}
+          cache: poetry
+
+      - name: Install dependencies
+        run: poetry install --only=release
+
+      - name: Install pandoc
+        run: sudo apt update && sudo apt install pandoc -y
+
+      - name: Obtain the changelog
+        run: echo "changelog=$(poe changelog)" >> $GITHUB_OUTPUT
+
       - name: Build a binary wheel and a source tarball
-        env:
-          TZ: UTC
-        run: python3 -m build
+        run: poetry build
+
       - name: Store the distribution packages
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
           name: python-package-distributions
           path: dist/
 
-  make_github_release:
+  publish-and-release:
+    name: Create GitHub release and publish distribution 📦 to PyPI
     runs-on: ubuntu-latest
     needs: build
+    env:
+      CHANGELOG: ${{ needs.build.outputs.changelog }}
+    environment:
+      name: pypi
+      url: https://pypi.org/p/beets
+    permissions:
+      id-token: write
```

Comment:
Are there any issues, security or otherwise, with making this permission wider? Just want to be sure, because before it was only applied to the PyPi release action, now it's for a lot of different steps.

Reply:
This is a very very very good point, thank you! In my world security always wins over anything else, thus I will undo the last commit